### DISCUSSION TITLE
Browse Reponse TypeDefinition

### DISCUSSION THREAD

**Discussion by AndreasHeine** [FreeOpcUa/opcua-asyncio#540] [General]
@oroulet @swamper123 

**TypeDefinition:**
ReferenceDescription(
ReferenceTypeId=NodeId(Identifier=**40**, NamespaceIndex=0, NodeIdType=<NodeIdType.TwoByte: 0>), 
IsForward=True, 
NodeId=**NodeId(Identifier=1006, NamespaceIndex=4, NodeIdType=<NodeIdType.Numeric: 2>)**, 
BrowseName=QualifiedName(NamespaceIndex=4, Name='SurfaceTechnologyObjectType'), 
DisplayName=LocalizedText(Locale=None, Text='SurfaceTechnologyObjectType'), 
NodeClass_=<NodeClass.DataType: 64>, 
TypeDefinition=**ExpandedNodeId(Identifier=0, NamespaceIndex=0, NodeIdType=<NodeIdType.TwoByte: 0>**, NamespaceUri=None, ServerIndex=0)), 

Why is TypeDefinition an empty ExpandedNodeId?

actually all if i dump them in a file:
```python
    with open("nodes.txt", "w") as f:
        for node in server.iserver.aspace._nodes:
            f.write(f"NodeId: {node} \n")
            f.write(f"References: {server.iserver.aspace._nodes[node].references}\n")
            f.write(f"-----------------------------------------------------------------------------\n")
```

**Comment by AndreasHeine** [comment]
currently looking in: https://github.com/FreeOpcUa/opcua-asyncio/blob/750ef8956f0845ecb6c6a1817fa6128df10cbcfb/asyncua/common/instantiate_util.py#L15

**Reply by oroulet** [reply]
sending an emtpy/null NodeId make the server create a new nodeid. using the next free nodeid in that namespace.
This is not part of spec but an extension I implemented
